.github/actions/people/people.py

```diff
@@ -0,0 +1,613 @@
+    # authors = {**issues_authors, **discussions_authors}
+    authors = {**discussions_authors}
+    return commentors, last_month_commentors, authors
+
+
+def get_contributors(settings: Settings):
+    pr_nodes: List[PullRequestNode] = []
+    pr_edges = get_graphql_pr_edges(settings=settings)
+
+    while pr_edges:
+        for edge in pr_edges:
+            pr_nodes.append(edge.node)
+        last_edge = pr_edges[-1]
+        pr_edges = get_graphql_pr_edges(settings=settings, after=last_edge.cursor)
+
+    contributors = Counter()
+    commentors = Counter()
+    reviewers = Counter()
+    authors: Dict[str, Author] = {}
+
+    for pr in pr_nodes:
+        author_name = None
+        if pr.author:
+            authors[pr.author.login] = pr.author
+            author_name = pr.author.login
+        pr_commentors: Set[str] = set()
+        pr_reviewers: Set[str] = set()
+        for comment in pr.comments.nodes:
+            if comment.author:
+                authors[comment.author.login] = comment.author
+                if comment.author.login == author_name:
+                    continue
+                pr_commentors.add(comment.author.login)
+        for author_name in pr_commentors:
+            commentors[author_name] += 1
+        for review in pr.reviews.nodes:
+            if review.author:
+                authors[review.author.login] = review.author
+                pr_reviewers.add(review.author.login)
+        for reviewer in pr_reviewers:
+            reviewers[reviewer] += 1
+        if pr.state == "MERGED" and pr.author:
+            contributors[pr.author.login] += 1
+    return contributors, commentors, reviewers, authors
+
+
+def get_top_users(
+    *,
+    counter: Counter,
+    min_count: int,
+    authors: Dict[str, Author],
+    skip_users: Container[str],
+):
+    users = []
+    for commentor, count in counter.most_common(50):
+        if commentor in skip_users:
+            continue
+        if count >= min_count:
+            author = authors[commentor]
+            users.append(
+                {
+                    "login": commentor,
+                    "count": count,
+                    "avatarUrl": author.avatarUrl,
+                    "url": author.url,
+                }
+            )
+    return users
+
+
+if __name__ == "__main__":
+    logging.basicConfig(level=logging.INFO)
+    settings = Settings()
+    logging.info(f"Using config: {settings.model_dump_json()}")
+    g = Github(settings.input_token.get_secret_value())
+    repo = g.get_repo(settings.github_repository)
+    question_commentors, question_last_month_commentors, question_authors = get_experts(
+        settings=settings
+    )
+    contributors, pr_commentors, reviewers, pr_authors = get_contributors(
+        settings=settings
+    )
+    authors = {**question_authors, **pr_authors}
+    maintainers_logins = {'samuelcolvin', 'adriangb', 'dmontagu', 'hramezani', 'lig', 'Kludex', 'davidhewitt', 'sydney-runkle'}
```

Comment:
Should we still keep `lig` as the maintainer?

Reply:
I'm guessing not, but we can leave him as an author in `pyproject.toml`...